SU2_CFD/src/output/filewriter/CCGNSFileWriter.cpp

```diff
@@ -0,0 +1,306 @@
+    if(cgns_err) cg_error_print;
+    cell_dim = 3;
+    phys_dim = 3;
+    cgns_err = cg_base_write(cgns_file1, basename, cell_dim, phys_dim,  &base_number);
+    if(cgns_err) cg_error_print;
+    cgns_err = cg_zone_write(cgns_file1, base_number, zonename, *isize, Unstructured, &zone_number);
+
+    /*---    Write Coordinates    ---*/
+    cgns_err = cg_ncoords(cgns_file, nbases, nzones, &ncoords);
+    if(cgns_err) cg_error_print();
+    for (i = 1; i < ncoords + 1; i++){
+        cgns_err = cg_coord_info(cgns_file, nbases, nzones, i, &datatype, coordname);
+        if(cgns_err) cg_error_print();
+        start = 1;
+        end = isize[0][0];
+        vector<double> buf(end);
+        cgns_err = cg_coord_read(cgns_file, nbases, nzones, coordname,
+                                 datatype, &start, &end, buf.data());
+        if(cgns_err) cg_error_print();
+        cgns_err = cg_coord_write(cgns_file1, base_number, zone_number, datatype, coordname,
+                                  buf.data(), &coord_index);
+        if(cgns_err) cg_error_print();
+    }
+
+    /*---    Write Connectivity    ---*/
+    cgns_err = cg_nsections(cgns_file, nbases, nzones, &nsections);
+    if(cgns_err) cg_error_print();
+    for ( i = 1; i < nsections + 1; i++)    {
+        cgns_err = cg_section_read(cgns_file, nbases, nzones, i,
+                                   ElementSectionName, &type, &start,
+                                   &end, &nbndry, &parent_flag);
+        if(cgns_err) cg_error_print();
+        cgns_err = cg_ElementDataSize(cgns_file, nbases, nzones, i, &ElementDataSize);
+        if(cgns_err) cg_error_print();
+        if(type != MIXED){
+            Elements = new cgsize_t[ElementDataSize];
+            cgns_err = cg_elements_read(cgns_file, nbases, nzones, i, Elements, NULL);
+            if(cgns_err) cg_error_print();
+            cgns_err = cg_section_write(cgns_file1, base_number, zone_number,
+                                        ElementSectionName, type, start,
+                                        end, nbndry, Elements, &S);
+            if(cgns_err) cg_error_print();
+        }
+        else{
+            Elements = new cgsize_t[ElementDataSize];
+            cgns_err = cg_poly_elements_read(cgns_file, nbases, nzones, i, Elements, ConnectOffset, NULL);
+            if(cgns_err) cg_error_print();
+            cgns_err = cg_poly_section_write(cgns_file1, base_number, zone_number,
+                                             ElementSectionName, type, start,
+                                             end, nbndry, Elements, ConnectOffset, &S);
+            if(cgns_err) cg_error_print();
+        }
+        /*---    Add Boundary Condition    ---*/
+        for (j = 0; j <= config->GetnMarker_All(); j++){
+            if (ElementSectionName == config->GetMarker_All_TagBound(j)){
+                if (config->GetMarker_All_KindBC(j) == 26){
+                    bocotype = (BCType_t)20;
```

Comment:
Can these constants, and a few below, be given names as well? like Mick suggested for "MIXED"

Reply:
Yes, I gave them a name declaring new variables. Now it should be more clear.